File: modules/opla-front/src/app/mlconfig/mlmodels/KStarLearningModel.ts
import {MachineLearningModel} from "./MachineLearningModel";
import {MachineLearningAlgorithm} from "./MachineLearningAlgorithm";

export class KStarLearningModel extends MachineLearningModel{

  entropicAutoBlend: boolean;
  globalBlend: string;
  missingMode: string;

  constructor(entropicAutoBlend: string, globalBlend: string, missingMode: string) {
    super(MachineLearningAlgorithm.KSTAR);
    this.entropicAutoBlend = (entropicAutoBlend === "true");
    this.globalBlend = globalBlend;
    this.missingMode = missingMode;
  }
}

File: modules/opla-core/src/main/java/br/otimizes/oplatool/core/learning/mlmodels/KStarLearningModel.java
package br.otimizes.oplatool.core.learning.mlmodels;

import weka.core.SelectedTag;

public class KStarLearningModel {
    private boolean entropicAutoBlend;
    private int globalBlend;
    private SelectedTag missingMode;

    public KStarLearningModel(boolean entropicAutoBlend, int globalBlend, SelectedTag missingMode) {
        this.entropicAutoBlend = entropicAutoBlend;
        this.globalBlend = globalBlend;
        this.missingMode = missingMode;
    }

    public boolean isEntropicAutoBlend() {
        return entropicAutoBlend;
    }

    public void setEntropicAutoBlend(boolean entropicAutoBlend) {
        this.entropicAutoBlend = entropicAutoBlend;
    }

    public int getGlobalBlend() {
        return globalBlend;
    }

    public void setGlobalBlend(int globalBlend) {
        this.globalBlend = globalBlend;
    }

    public SelectedTag getMissingMode() {
        return missingMode;
    }

    public void setMissingMode(SelectedTag missingMode) {
        this.missingMode = missingMode;
    }
}
